packages/itwin/measure-tools/src/tools/MeasureAreaTool.ts

```diff
@@ -61,13 +63,34 @@ MeasureAreaToolModel
     this._enableSheetMeasurements = enableSheetMeasurements;
   }
 
+  private async updateDrawingTypeCache() {
+    this._sheetChangeListener();
+    const sheetIds = [];
+    if (this._enableSheetMeasurements) {
+      for (const viewport of IModelApp.viewManager) {
+        if (viewport.view.classFullName === "BisCore:SheetViewDefinition") {
+          this._sheetChangeListener = viewport.onViewedModelsChanged.addListener(async () => this.updateDrawingTypeCache());
+          sheetIds.push(viewport.view.id);
+        }
+      }
+    }
+
+    for (const id of sheetIds) {
+      this._drawingTypeCache = await SheetMeasurementsHelper.getSheetTypes(this.iModel, id);
+    }
+  }
+
   public async onRestartTool(): Promise<void> {
     const tool = new MeasureAreaTool(this._enableSheetMeasurements);
     if (await tool.run()) return;
 
     return this.exitTool();
   }
 
+  public override async onPostInstall(): Promise<void> {
+    await this.updateDrawingTypeCache();
```

Comment:
Not sure if base class does anything but I'd add `await super.onPostInstall()` for safety.